### PR TITLE
Create a new h2 stream only if the parent channel is still active

### PR DESCRIPTION
**Pull request by idelpivnitskiy**
Motivation:

If a request to open a new h2 stream was made from outside of the
EventLoop it will be scheduled for future execution on the EventLoop.
However, during the time before the `open0` task will be executed the
parent channel may already be closed. As a result,
`Http2MultiplexHandler#newOutboundStream()` will throw an
`IllegalStateException` with the message that is hard to
interpret correctly for this use-case: "Http2FrameCodec not found. Has
the handler been added to a pipeline?".

Modifications:

- Check that the parent h2 `Channel` is still active before creating a
new stream when `open0` task is picked up by EventLoop;

Result:

Users see a correct `ClosedChannelException` in case the parent h2
`Channel` was closed concurrently with a request for a new stream.

### CODE REVIEW

**Comment by normanmaurer** [comment]
@idelpivnitskiy can you add a unit test as well ?

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by idelpivnitskiy** [comment]
@normanmaurer added unit-test, but not sure if the approach for the test is right, ptal.

File: codec-http2/src/test/java/io/netty/handler/codec/http2/Http2StreamChannelBootstrapTest.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
+package io.netty.handler.codec.http2;
```
Comment (normanmaurer) [inline codec-http2/src/test/java/io/netty/handler/codec/http2/Http2StreamChannelBootstrapTest.java:1]: nit: add license header

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Pull request merged**